Formula/dafny.rb

```diff
@@ -0,0 +1,41 @@
+class Dafny < Formula
+  desc "Verification-aware programming language"
+  homepage "https://www.microsoft.com/en-us/research/project/dafny-a-language-and-program-verifier-for-functional-correctness"
+  url "https://github.com/dafny-lang/dafny/archive/v2.3.0.tar.gz"
+  sha256 "ea7ae310282c922772a46a9a85e2b4213043283038b74d012047b5294687d168"
+
+  depends_on "mono"
+  depends_on "mono-libgdiplus"
+  depends_on "nuget"
+  depends_on "z3"
+
+  resource "boogie" do
+    url "https://github.com/boogie-org/boogie.git"
```

Comment:
This needs to be a tagged release as well.

Reply:
there is not tagged release for this, so just added the default branch to this

Reply:
There needs to be a locked target for resources otherwise every build is going to be different from the last and that's impossible to debug.

Reply:
Okay, I see your point. Just added specific revision